docs/extra/k8s-cheatsheet.md

```diff
@@ -0,0 +1,98 @@
+# Kubernetes Cheatsheet
+
+## Session Manager SSH
+
+When using AWS Session Manager since the CLI is very limited you will benefit from using these aliases:
+```bash
+alias kubectl='sudo microk8s.kubectl -n codacy'
+alias helm='sudo helm'
+```
+
+## How to install a custom Codacy version
+
+### Install
+
+```bash
+sudo git clone git://github.com/codacy/chart -b <YOUR-BRANCH>
+helm dep build ./chart/codacy
+helm upgrade --install codacy ./chart/codacy/ --namespace codacy --atomic --timeout=300 --values ./<YOUR-VALUES-FILE>
+```
+
+### Update
+
+```bash
+(cd chart; sudo git fetch --all --prune --tags; sudo git reset --hard origin/<YOUR-BRANCH>;)
+helm dep build ./chart/codacy
+helm upgrade --install codacy ./chart/codacy/ --namespace codacy --atomic --timeout=300 --values ./<YOUR-VALUES-FILE>
+```
+
+## How to clean the namespace
+
+```
+helm del --purge codacy
+kubectl -n codacy delete --all pod &
+kubectl -n codacy delete --all pvc &
+kubectl -n codacy delete --all pv  &
+kubectl -n codacy delete --all job &
+sleep 5
+kubectl -n codacy patch pvc -p '{"metadata":{"finalizers":null}}' $(kubectl -n codacy get pvc -o jsonpath='{.items[*].metadata.name}')
+kubectl -n codacy patch pv -p '{"metadata":{"finalizers":null}}' $(kubectl -n codacy get pv -o jsonpath='{.items[*].metadata.name}')
+sleep 5
+kubectl -n codacy delete pod $(kubectl -n codacy get pod -o jsonpath='{.items[*].metadata.name}') --force --grace-period=0
+kubectl -n codacy get pod &
+kubectl -n codacy get pvc &
+kubectl -n codacy get pv  &
+kubectl -n codacy get job &
+```
+
+### Check uninstall process
+ps aux | grep -i kubectl
+
+## Get logs of a service
+
+```bash
+kubectl get services
+kubectl logs svc/<service-name>
+```
+
+## Edit configmap
+
+```bash
+kubectl get configmaps
+kubectl edit configmap <configmap-name>
+```
+
+## Restart deployment of daemonset
+
+```bash
+kubectl get daemonsets
+kubectl rollout restart daemonset/<daemonset-name>
+
+kubectl get deployment
+kubectl rollout restart deployment/<deployment-name>
+kubectl rollout status deployment/<deployment-name> -w
+```
+
+## Read logs
+
+### daemonset with multiple containers
+
+```bash
+kubectl logs daemonset/<daemonset-name> <container-name> -f
+```
+
+### service
+
+```bash
+kubectl logs service/<service-name> -f
+```
+
+## Open shell inside container
+
+```bash
+kubectl exec -it daemonset/<daemonset-name> -c <container-name> sh
+
+# or
+
+kubectl exec -it deployment/<deployment-name> sh
+```
```

Comment:
Separate these into two bash codeblocks? I think the "or" will be clearer / more visible.

Reply:
Any special reason why? The reason you have a code block is usually to have multiple code blocks.

Reply:
The user will run either one command or the other, correct? So it will make it easier to copy + paste only the intended command.

Reply:
These are just examples. You can copy each individually.
None of these should be run together.
These are all commands you need to tweak to do something.

In that case the idea is to show one where we have multiple containers and another when you don't.

Reply:
Exactly... The MkDocs template we're using includes a button that allows copying the content of codeblocks for convenience (you can see this working [here](https://codacy.github.io/chart/install/)):

![image](https://user-images.githubusercontent.com/60105800/77078957-3675a000-69ef-11ea-9fb3-6f336efd31dc.png)

What I was trying to say is that if we have separate codeblocks for each command that the user must choose from, it becomes more user friendly.

Reply:
ok, that sounds like a good reason